src/Cody.VisualStudio/Services/AgentService.cs

```diff
@@ -0,0 +1,134 @@
+using Cody.Core.Agent;
+using Cody.Core.Agent.Protocol;
+using Cody.Core.Logging;
+using Cody.VisualStudio.Client;
+using Microsoft.VisualStudio.Shell;
+using System;
+using System.Threading.Tasks;
+
+namespace Cody.VisualStudio.Services
+{
+    public class AgentService: IDisposable
```

Comment:
this name conflicts with `IAgentService` and the related field. maybe this class should be named `AgentLifecycleManager`?

Reply:
I've renamed it already.